projects/drivers/xilinx_platform/platform_drivers.c

```diff
@@ -40,10 +40,87 @@
 /******************************************************************************/
 /***************************** Include Files **********************************/
 /******************************************************************************/
+#include <stdint.h>
 #include <stdlib.h>
 #include <sleep.h>
+#include <xparameters.h>
+#ifdef _XPARAMETERS_PS_H_
+#include <xspips.h>
+#include <xgpiops.h>
+#include <xiic.h>
+#include <xil_exception.h>
+#else
+#include <xspi.h>
+#include <xgpio.h>
+#endif
 #include "platform_drivers.h"
 
+/******************************************************************************/
+/********************** Macros and Constants Definitions **********************/
+/******************************************************************************/
+
+#define SPI_CS_DECODE	0x01
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+typedef enum i2c_type {
+	XILINX_I2C
+} i2c_type;
```

Comment:
This structure seems to be useless. Same as spi_type and gpio_type.

Reply:
These enumerations are used as attributes inside the descriptor/init_param structures. I suppose their purpose is to support extensions for multiple i2c types.